src/librustc/ty/maps.rs

```diff
@@ -107,13 +118,25 @@ impl<'tcx> Value<'tcx> for Ty<'tcx> {
     }
 }
 
-
 impl<'tcx> Value<'tcx> for ty::DtorckConstraint<'tcx> {
     fn from_cycle_error<'a>(_: TyCtxt<'a, 'tcx, 'tcx>) -> Self {
         Self::empty()
     }
 }
 
+impl<'tcx> Value<'tcx> for SymbolName {
+    fn from_cycle_error<'a>(_: TyCtxt<'a, 'tcx, 'tcx>) -> Self {
+        SymbolName { name: Symbol::intern("<error>").as_str() }
+    }
+}
+
+#[derive(Clone)]
+pub struct SymbolName {
+    // FIXME: we don't rely on interning or equality here - better have
+    // this be a `&'tcx str`.
```

Comment:
What do you mean by "don't rely on interning or equality here"? Maybe you mean *for* equality? The main thing to be careful of (which `&'tcx str` would solve) is that we hash the string contents (if you use `Symbol`, things go awry).